clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -466,7 +466,8 @@ public void poll(long timeout) throws IOException {
         long startSelect = time.nanoseconds();
         int numReadyKeys = select(timeout);
         long endSelect = time.nanoseconds();
-        log.trace("Completed select in " + (endSelect - startSelect) + "ns");
+        log.trace("Completed select in " + (endSelect - startSelect) + "ns, with numReadyKeys " + numReadyKeys + ", "
```

Comment:
As suggested by Greg, we should use the slf4j {} substitution to avoid the cost of string construction when the trace level logging is not enabled.